File: crm_dashboard/src/lib/labels.ts
// Backend enum values → what the user actually reads. One place, so the
// same status can't end up as "Menang" on one screen and "Won" on
// another (Aturan #12: seluruh teks antarmuka Bahasa Indonesia).
//
// Labels come from the Claude Design output (issue #40); colors and
// shapes from the Phase 8.6 token sheet (issue #159). The colors are NOT
// CSS tokens: each one belongs to exactly one enum value rather than to
// the theme, and Tailwind can't generate classes from a runtime value
// anyway — these are applied as inline styles by whatever renders them.

// --- Lead status -------------------------------------------------------

export const LEAD_STATUSES = [
  "new",
  "contacted",
  "qualified",
  "proposal",
  "won",
  "lost",
  "unqualified",
  "spam",
] as const;

export type LeadStatus = (typeof LEAD_STATUSES)[number];

/**
 * How a status badge is drawn on the dashboard (token sheet "Opsi B").
 * The shape carries meaning so status never rests on color alone:
 *   pill   — the main path (Baru → Penawaran)
 *   square — an outcome (Menang, Kalah)
 *   dashed — excluded from conversion rate (Tidak Memenuhi Syarat, Spam)
 */
export type StatusShape = "pill" | "square" | "dashed";

export interface StatusMeta {
  label: string;
  /** Text and border color. Drawn on white in the badge itself. */
  color: string;
  /** Tint for surfaces that carry this status's text on a colored
   * background (closed-lead ribbon, active filter chip). */
  background: string;
  shape: StatusShape;
}

// Contrast, recomputed rather than copied (issue #159 — table in
// docs/phases/08.6-ui-redesign/notes.md): every `color` is 5.03–5.27:1
// on white, which is what the badge draws on, and the same for white
// text on `color` (an active chip).
//
// The token sheet claims these reach ≥4.6:1 "on their own tint". They
// don't at the old 85%-white tint (4.11–4.29:1), so `background` is
// mixed at 94% white instead: 4.65:1 at the worst (`won`). That is what
// AA needs for normal text wherever a status color sits on its tint.
const tint = (color: string) => `color-mix(in oklch, ${color}, white 94%)`;

const status = (label: string, color: string, shape: StatusShape): StatusMeta => ({
  label,
  color,
  background: tint(color),
  shape,
});

export const STATUS_META: Record<LeadStatus, StatusMeta> = {
  new: status("Baru", "#006cd3", "pill"),
  contacted: status("Dihubungi", "#8156c0", "pill"),
  qualified: status("Memenuhi Syarat", "#007b7d", "pill"),
  proposal: status("Penawaran", "#a05d00", "pill"),
  won: status("Menang", "#1d802b", "square"),
  lost: status("Kalah", "#ce2930", "square"),
  unqualified: status("Tidak Memenuhi Syarat", "#6d6d6d", "dashed"),
  spam: status("Spam", "#7f6964", "dashed"),
};

// --- Lost reason -------------------------------------------------------

export const LOST_REASONS = [
  "price",
  "competitor",
  "timing",
  "no_response",
  "not_interested",
  "other",
] as const;

export type LostReason = (typeof LOST_REASONS)[number];

export const LOST_REASON_LABELS: Record<LostReason, string> = {
  price: "Harga",
  competitor: "Kompetitor",
  timing: "Waktu Tidak Tepat",
  no_response: "Tidak Merespons",
  not_interested: "Tidak Tertarik",
  other: "Lainnya",
};

// --- Lead source -------------------------------------------------------

export const LEAD_SOURCES = ["manual", "api", "form", "webhook"] as const;

export type LeadSource = (typeof LEAD_SOURCES)[number];

// "Formulir", not "Form" — this is the one source with a natural
// Indonesian word. API and Webhook stay as-is; they're proper nouns to
// the integrator who sees them.
export const SOURCE_LABELS: Record<LeadSource, string> = {
  manual: "Manual",
  api: "API",
  form: "Formulir",
  webhook: "Webhook",
};

// --- Role --------------------------------------------------------------

export const ROLES = ["owner", "admin", "manager", "employee"] as const;

export type Role = (typeof ROLES)[number];

// Left in English on purpose: these are the exact terms glossary.md
// fixes for the role enum, and the product speaks about "Owner" and
// "Admin" that way throughout.
export const ROLE_LABELS: Record<Role, string> = {
  owner: "Owner",
  admin: "Admin",
  manager: "Manager",
  employee: "Employee",
};

// --- API key scope -------------------------------------------------------

// Only one scope exists in this codebase today (ADR-004 aturan #4) — a
// Record over the full APIKeyScope union, not a partial map, so adding a
// second scope value without a label is a compile error, not a blank
// cell in the UI.
export const SCOPE_LABELS: Record<"leads:write", string> = {
  "leads:write": "Kirim lead",
};

File: crm_employee/lib/shared/labels.dart
import 'package:flutter/material.dart';

/// Backend enum values → what the user actually reads. Copied from
/// `crm_dashboard/src/lib/labels.ts`'s values (TD §3 — "menyalin nilai,
/// bukan mengimpornya": no Dart↔TypeScript code-sharing mechanism
/// exists, and building one would be an abstraction for a problem that
/// doesn't exist, Aturan #27). `glossary.md` stays the one source of
/// truth for the terms themselves; `test/shared/labels_test.dart` locks
/// each list to the same length as the backend enum (Aturan #12: seluruh
/// teks antarmuka Bahasa Indonesia).
///
/// Deliberately does NOT include `SCOPE_LABELS` (API key scopes) —
/// `crm_dashboard`'s labels.ts has that map because the dashboard manages
/// API keys; `crm_employee` never touches the API key format at all
/// (Aturan #24), so porting that map here would be the one piece of
/// `jln_*`-adjacent vocabulary this app has no business knowing.

// --- Lead status -----------------------------------------------------

const List<String> leadStatuses = [
  'new',
  'contacted',
  'qualified',
  'proposal',
  'won',
  'lost',
  'unqualified',
  'spam',
];

/// How a status is drawn in this app — the token sheet's "Opsi A" (Phase
/// 8.6, #172): an icon plus the label, in [color] on its own [tint]. On a
/// one-handed phone in the sun the icon is read before the text, so shape
/// carries the status and color never does alone. (The dashboard uses
/// "Opsi B", outline + shape: in a 25-row table eight icons turn into
/// noise.) Hues match the dashboard's, so a status looks like itself on
/// both.
///
/// Every [color]/[tint] pair is 7.01–7.13:1 — recomputed, and re-checked by
/// test/shared/theme_contrast_test.dart.
@immutable
class StatusMeta {
  final String label;

  /// Text and icon color.
  final Color color;

  /// Badge background.
  final Color tint;

  final IconData icon;

  const StatusMeta({
    required this.label,
    required this.color,
    required this.tint,
    required this.icon,
  });
}

const Map<String, StatusMeta> statusMeta = {
  'new': StatusMeta(
    label: 'Baru',
    color: Color(0xFF004EB3),
    tint: Color(0xFFEDF8FF),
    icon: Icons.add_circle_outline,
  ),
  'contacted': StatusMeta(
    label: 'Dihubungi',
    color: Color(0xFF673BA2),
    tint: Color(0xFFF9F4FF),
    icon: Icons.phone_outlined,
  ),
  'qualified': StatusMeta(
    label: 'Memenuhi Syarat',
    color: Color(0xFF005E60),
    tint: Color(0xFFEBFAFA),
    icon: Icons.check,
  ),
  'proposal': StatusMeta(
    label: 'Penawaran',
    color: Color(0xFF834300),
    tint: Color(0xFFFFF5E9),
    icon: Icons.description_outlined,
  ),
  'won': StatusMeta(
    label: 'Menang',
    color: Color(0xFF006307),
    tint: Color(0xFFEFFBEF),
    icon: Icons.star_outline,
  ),
  'lost': StatusMeta(
    label: 'Kalah',
    color: Color(0xFFAB000E),
    tint: Color(0xFFFFF2EF),
    icon: Icons.close,
  ),
  // The two statuses excluded from conversion rate are both near-gray on
  // purpose — so their ICONS must differ: "minus" (not a fit) vs "ban"
  // (junk). Color can't tell them apart (ΔE 3 on the dashboard, #171).
  'unqualified': StatusMeta(
    label: 'Tidak Memenuhi Syarat',
    color: Color(0xFF545454),
    tint: Color(0xFFF7F7F7),
    icon: Icons.remove_circle_outline,
  ),
  'spam': StatusMeta(
    label: 'Spam',
    color: Color(0xFF654F4B),
    tint: Color(0xFFF9F6F5),
    icon: Icons.block,
  ),
};

// --- Lost reason -------------------------------------------------------

const List<String> lostReasons = [
  'price',
  'competitor',
  'timing',
  'no_response',
  'not_interested',
  'other',
];

const Map<String, String> lostReasonLabels = {
  'price': 'Harga',
  'competitor': 'Kompetitor',
  'timing': 'Waktu Tidak Tepat',
  'no_response': 'Tidak Merespons',
  'not_interested': 'Tidak Tertarik',
  'other': 'Lainnya',
};

// --- Lead source ---------------------------------------------------------

const List<String> leadSources = ['manual', 'api', 'form', 'webhook'];

// "Formulir", not "Form" — the one source with a natural Indonesian
// word. API and Webhook stay as-is; they're proper nouns to the
// integrator who sees them (same reasoning as labels.ts).
const Map<String, String> sourceLabels = {
  'manual': 'Manual',
  'api': 'API',
  'form': 'Formulir',
  'webhook': 'Webhook',
};

// --- Role ----------------------------------------------------------------

const List<String> roles = ['owner', 'admin', 'manager', 'employee'];

// Left in English on purpose — glossary.md fixes these exact terms for
// the role enum, and the product speaks about "Owner"/"Admin" that way
// throughout, same as labels.ts.
const Map<String, String> roleLabels = {
  'owner': 'Owner',
  'admin': 'Admin',
  'manager': 'Manager',
  'employee': 'Employee',
};
